daemon/exec.go

```diff
@@ -248,3 +248,34 @@ func (d *Daemon) Exec(c *Container, execConfig *execConfig, pipes *execdriver.Pi
 
 	return exitStatus, err
 }
+
+// execCommandGC starts a go routine on a ticker to clean up the daemon references
+// of exec configs that are no longer part of the container.
+func (d *Daemon) execCommandGC() {
+	for range time.Tick(5 * time.Minute) {
+		var (
+			cleaned          int
+			liveExecCommands = d.containerExecIds()
+			ids              = d.execCommands.List()
+		)
+		for _, id := range ids {
+			if _, exists := liveExecCommands[id]; !exists {
```

Comment:
i actually think the time based approach is better and it's much simpler, we can iterate on this later if it turns out to be a bad idea.  since we are already timebased on the GC we will just not remove them if they exited less than 60 seconds before the gc loop.